.github/workflows/publish_maven_central.yml

```
@@ -0,0 +1,37 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Publish package to Maven central
+
+on:
+  push:
+    tags: v*
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: zendesk/checkout@v2
+    - name: Set up JDK 1.8
+      uses: zendesk/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Grant execute permission for gradlew
+      run: chmod +x gradlew
+    - name: Prepare environment
+      env:
+        GPG_KEY_CONTENTS: ${{ secrets.GPG_KEY_CONTENTS }}
+        SIGNING_SECRET_KEY_RING_FILE: ${{ secrets.SIGNING_SECRET_KEY_RING_FILE }}
```

This does not need to be secret IMO, given we are creating this file with this action anyway
```suggestion
        SIGNING_SECRET_KEY_RING_FILE: ~/.keyring.gpg
```